packages/nestjs/src/backtrace.handler.ts

```diff
@@ -0,0 +1,208 @@
+        this._client = client;
+    }
+
+    public handleException(err: unknown, host: Context) {
+        const client = this._client ?? BacktraceClient.instance;
+        if (!client) {
+            if (this._options.skipIfClientUndefined) {
+                return false;
+            }
+
+            throw new Error('Backtrace instance is unavailable. Initialize the client first.');
+        }
+
+        if (!this.shouldSend(err)) {
+            return false;
+        }
+
+        let attributes: Record<string, unknown> = {
+            ...this.getBaseAttributes(host),
+            ...this.getTypeAttributes(host),
+        };
+
+        if (this._options.buildAttributes) {
+            attributes = this._options.buildAttributes(host, attributes);
+        }
+
+        if (typeof err !== 'string' && !(err instanceof Error)) {
+            client.send(String(err), attributes);
+        } else {
+            client.send(err, attributes);
+        }
+
+        return true;
+    }
+
+    private shouldSend(error: unknown) {
+        if (this._options.includeExceptionTypes && this.filterException(error, this._options.includeExceptionTypes)) {
+            return true;
+        }
+
+        if (this._options.excludeExceptionTypes && this.filterException(error, this._options.excludeExceptionTypes)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private getBaseAttributes(host: ArgumentsHost) {
+        const contextType = host.getType();
+
+        return {
+            'request.contextType': contextType,
+        };
+    }
+
+    private getTypeAttributes(host: ArgumentsHost) {
+        const type = host.getType();
+        switch (type) {
+            case 'http':
+                return this.getHttpAttributes(host.switchToHttp());
+            case 'rpc':
+                return this.getRpcAttributes(host.switchToRpc());
+            case 'ws':
+                return this.getWsAttributes(host.switchToWs());
+            default:
+                return {};
+        }
+    }
+
+    private getHttpAttributes(http: HttpArgumentsHost) {
+        const request = http.getRequest();
+        const expressRequest = request as ExpressRequest;
+        return {
+            'request.url': expressRequest.url,
+            'request.baseUrl': expressRequest.baseUrl,
+            'request.method': expressRequest.method,
+            'request.originalUrl': expressRequest.originalUrl,
+            'request.protocol': expressRequest.protocol,
+            'request.hostname': expressRequest.hostname,
+            'request.httpVersion': expressRequest.httpVersion,
+        };
+    }
+
+    private getRpcAttributes(rpc: RpcArgumentsHost) {
+        return {
+            ['rpc.data']: rpc.getData(),
+        };
+    }
+
+    private getWsAttributes(ws: WsArgumentsHost) {
+        return {
+            ['ws.data']: ws.getData(),
+        };
+    }
+
+    private filterException(exception: unknown, filter: ExceptionTypeFilter): boolean {
```

Comment:
you're not filtering exception but rather you're checking if the exception match filter criteria. It was hard for me to check if you have a bug in the "sendError" method, because you used the same way the filterException method. 

Maybe a different name can improve readability? for example: `filterIncludeError` ?

Reply:
I renamed it to `testException` and added some docs.